File: DriftGame/uracer-desktop/src/com/bitfire/uracer/utils/BoxedFloatAccessor.java
package com.bitfire.uracer.utils;

import aurelienribon.tweenengine.TweenAccessor;

public final class BoxedFloatAccessor implements TweenAccessor<BoxedFloat> {
    public static final int VALUE = 1;

    @Override
    public int getValues(BoxedFloat target, int tweenType, float[] returnValues) {
        returnValues[0] = target.value;
        return 1;
    }

    @Override
    public void setValues(BoxedFloat target, int tweenType, float[] newValues) {
        target.value = newValues[0];
    }
}

File: DriftGame/uracer-desktop/src/com/bitfire/uracer/utils/BoxedFloatAccessor.kt
package com.bitfire.uracer.utils

import aurelienribon.tweenengine.TweenAccessor

class BoxedFloatAccessor : TweenAccessor<BoxedFloat> {

    override fun getValues(target: BoxedFloat, tweenType: Int, returnValues: FloatArray): Int {
        returnValues[0] = target.value
        return 1
    }

    override fun setValues(target: BoxedFloat, tweenType: Int, newValues: FloatArray) {
        target.value = newValues[0]
    }

    companion object {
        const val VALUE = 1
    }
}
